### PR TITLE
Avoid crawling trash

### PR DESCRIPTION
**Pull request by marcelklehr**
fixes https://github.com/nextcloud/recognize/issues/315

### CODE REVIEW

File: lib/BackgroundJobs/SchedulerJob.php [hunk 1]
```diff
@@ -24,28 +27,55 @@ class SchedulerJob extends QueuedJob {
...
+		$qb->select('root_id', 'storage_id', 'mount_provider_class')
 			->from('mounts')
 			->where($qb->expr()->in('mount_provider_class', $qb->createPositionalParameter(self::ALLOWED_MOUNT_TYPES, IQueryBuilder::PARAM_STR_ARRAY)));
 		$result = $qb->executeQuery();
 
 		while ($row = $result->fetch()) {
+			$storageId = (int)$row['storage_id'];
+			$rootId = (int)$row['root_id'];
+			$overrideRoot = $rootId;
+			if ($row['mount_provider_class'] === self::LOCAL_HOME_MOUNT_TYPE) {
```
Comment (PVince81) [inline lib/BackgroundJobs/SchedulerJob.php:59]: does this work also with s3 as primary storage or is there different handling ?

File: lib/BackgroundJobs/SchedulerJob.php [hunk 2]
```diff
@@ -24,28 +27,55 @@ class SchedulerJob extends QueuedJob {
...
 		$this->logger = $logger;
 		$this->db = $db;
 		$this->jobList = $jobList;
+		$this->systemConfig = $systemConfig;
 	}
 
 	protected function run($argument): void {
 		$qb = $this->db->getQueryBuilder();
-		$qb->select('root_id', 'storage_id')
+		$qb->select('root_id', 'storage_id', 'mount_provider_class')
```
Comment (PVince81) [inline lib/BackgroundJobs/SchedulerJob.php:50]: I wonder if there's a better way for doing all this through the APIs.
I'm guessing that using the APIs would be convoluted and would require iterating over all mounts and/or users just to find the file ids and maybe worse performance. @icewind1991 fyi

maybe we raise a tech debt ticket to address this later

**Review by PVince81 — APPROVED** [review]
:+1: if you can confirm that it also works with S3 primary storage

File: lib/BackgroundJobs/SchedulerJob.php (hunk 1 above)
Comment (marcelklehr) [reply lib/BackgroundJobs/SchedulerJob.php:59]: I'm not sure. How does s3 as primary storage work?

**Pull request merged**